pytorch_lightning/loops/dataloader/evaluation_dataloader_loop.py

```diff
@@ -0,0 +1,296 @@
+        self._max_batches: Optional[Union[int, Sequence[int]]] = None
+        self.outputs = []
+        self.evaluation_loop = EvaluationEpochLoop()
+
+        self._val_results = ResultCollection(training=False)
+        self._test_results = ResultCollection(training=False)
+
+    @property
+    def num_dataloaders(self) -> int:
+        """Returns the total number of dataloaders"""
+        return self._get_num_dataloaders(self.dataloaders)
+
+    @property
+    def dataloaders(self) -> Sequence[DataLoader]:
+        """Returns all dataloaders"""
+        return self._dataloaders
+
+    @property
+    def results(self) -> Optional[ResultCollection]:
+        """Returns the Results of obtained from all dataloaders"""
+        if self.trainer.validating or self.trainer.sanity_checking:
+            return self._val_results
+        elif self.trainer.testing:
+            return self._test_results
+        return None
+
+    @property
+    def predictions(self):
+        """Returns the predictions from all dataloaders"""
+        # TODO: fixme
+        return self.evaluation_loop.predictions
+
+    def connect(self, trainer: 'pl.Trainer', *args: Any, **kwargs: Any) -> None:
+        """Connects the loop to everything necessary (like trainer and accelerators)"""
+        super().connect(trainer, *args, **kwargs)
+
+        # TODO: why do we only forward *args and **kwargs here?
+        # TODO; Can we make the trainer a weakref/proxy?
+        self.evaluation_loop.connect(trainer, *args, **kwargs)
+
+    @property
+    def done(self) -> bool:
+        """Returns whether all dataloaders are processed or evaluation should be skipped altogether"""
+        return (self.current_dataloader_idx >= len(self.dataloaders)) or self.should_skip_evaluation(self._max_batches)
+
+    def reset(self) -> None:
+        """Resets the internal state of the loop"""
+        self.iteration_count = 0
+
+        # prepare dataloaders
+        self._dataloaders, self._max_batches = self.get_eval_dataloaders(), self.get_max_batches()
+        # bookkeeping
+        self.outputs = []
+
+        if isinstance(self._max_batches, int):
+            self._max_batches = [self._max_batches] * len(self._dataloaders)
+
+    def advance(self, *args: Any, **kwargs: Any) -> None:
+        """Performs evaluation on one single dataloader"""
+        void(*args, **kwargs)
+        dataloader = self.trainer.accelerator.process_dataloader(self.current_dataloader)
+        dataloader_iter = enumerate(dataloader)
+        dl_max_batches = self._max_batches[self.current_dataloader_idx]
+
+        dl_outputs = self.evaluation_loop.run(
+            dataloader_iter, self.current_dataloader_idx, dl_max_batches, self.num_dataloaders
+        )
+
+        # store batch level output per dataloader
+        if self.should_track_batch_outputs_for_epoch_end:
+            self.outputs.append(dl_outputs)
+
+    def on_run_start(self, *args: Any, **kwargs: Any) -> None:
```

Comment:
Could you re-order the functions in the call order. run, reset, on_run_start, etc... for all loops. It will be simpler to find the information.

Reply:
done. the only one I had to move was reset, the rest are already in order in all loops:

properties first
connect
run
reset
on_run_start
on_advance_start
advance
...
on_run_end
teardown